src/pyproject.rs

```diff
@@ -69,7 +70,10 @@ fn find_user_pyproject_toml() -> Option<PathBuf> {
 }
 
 fn find_project_root(sources: &[PathBuf]) -> Option<PathBuf> {
-    if let Some(prefix) = common_path_all(sources.iter().map(PathBuf::as_path)) {
+    let cwd = current_dir().unwrap_or_else(|_| ".".into());
+    // common_path doesn't work correctly with relative paths
+    let absolute_sources: Vec<PathBuf> = sources.iter().map(|source| cwd.join(source)).collect();
```

Comment:
Nit: `Vec` allocation is not necessary, as `common_path_all` accepts an iterator. I suggest merging `map` calls:

```rust
sources.iter().map(|source| cwd.join(source).as_path())
```

Reply:
`common_path_all` accepts an iterator of `Path` references. Your suggestion doesn’t compile as the temporary `PathBuf` isn’t in scope long enough.

Reply:
Oh indeed, I missed that. Sorry!